packages/frontend-core/src/fetch/DataFetch.js

```diff
@@ -176,10 +177,16 @@ export default class DataFetch {
       }
     }
 
+    let defaultQuery = {
+      logicalOperator: FilterGroupLogicalOperator.ALL,
+      groups: [],
+    }
+
     // Build the query
-    let query = this.options.query
-    if (!query) {
-      query = buildQuery(filter)
+    let query = this.options.query || null
+
+    if (!query && this.features.supportsSearch) {
```

Comment:
This was to act as a guard for any custom Dataprovider that may have been misconfigured with filters. The issue I mentioned above, where the filters were not cleared, would lead to an invalid query being used and the custom Dataprovider returning nothing.